samples/apps/copilot-chat-app/webapi/CopilotChat/Controllers/DocumentImportController.cs

```diff
@@ -76,108 +76,182 @@ private enum SupportedFileType
     /// Service API for importing a document.
     /// </summary>
     [Authorize]
-    [Route("importDocument")]
+    [Route("importDocuments")]
     [HttpPost]
     [ProducesResponseType(StatusCodes.Status200OK)]
     [ProducesResponseType(StatusCodes.Status400BadRequest)]
-    public async Task<IActionResult> ImportDocumentAsync(
+    public async Task<IActionResult> ImportDocumentsAsync(
         [FromServices] IKernel kernel,
         [FromServices] IHubContext<MessageRelayHub> messageRelayHubContext,
         [FromForm] DocumentImportForm documentImportForm)
     {
-        var formFile = documentImportForm.FormFile;
-        if (formFile == null)
+        try
         {
-            return this.BadRequest("No file was uploaded.");
+            await this.ValidateDocumentImportFormAsync(documentImportForm);
         }
+        catch (ArgumentException ex)
+        {
+            return this.BadRequest(ex.Message);
+        }
+
+        this._logger.LogInformation("Importing {0} document(s)...", documentImportForm.FormFiles.Count);
 
-        if (formFile.Length == 0)
+        // TODO: Perform the import in parallel.
+        DocumentMessageContent documentMessageContent = new();
+        foreach (var formFile in documentImportForm.FormFiles)
         {
-            return this.BadRequest("File is empty.");
+            var success = await this.ImportDocumentHelperAsync(kernel, formFile, documentImportForm);
+            documentMessageContent.AddDocument(
+                formFile.FileName,
+                this.GetReadableByteString(formFile.Length),
+                success);
         }
 
-        if (formFile.Length > this._options.FileSizeLimit)
+        // Broadcast the document uploaded event to other users.
+        if (documentImportForm.DocumentScope == DocumentImportForm.DocumentScopes.Chat)
         {
-            return this.BadRequest("File size exceeds the limit.");
+            var chatMessage = await this.TryCreateDocumentUploadMessage(
+                documentMessageContent,
+                documentImportForm);
+            if (chatMessage == null)
+            {
+                // It's Ok to have the message not created.
```

Comment:
When will the chat message be null? just when the files were not able to be imported?

Reply:
When a document is uploaded to the global scope, no message will be created.

Reply:
You are right. It should not be null. Returning a BadRequest instead.